colrev/ops/built_in/search_sources/eric.py

```diff
@@ -54,42 +76,183 @@ def validate_source(
-                f"File does not exist: query_file {source.search_parameters['query_file']} "
-                f"for ({source.filename})"
-            )
-
         search_operation.review_manager.logger.debug(
             f"SearchSource {source.filename} validated"
         )
-
     @classmethod
     def heuristic(cls, filename: Path, data: str) -> dict:
         """Source heuristic for ERIC"""
 
-        result = {"confidence": 0.0}
+        result = {"confidence": 0.1}
 
         # Note : no features in bib file for identification
 
         return result
+    
 
+    def search_split(search) -> str:
+        if ' AND ' in search:
+            search_parts = search.split(' AND ')
+            field_values = []
+            for part in search_parts:
+                field, value = part.split(':')
+                field = field.strip()
+                value = value.strip().strip("'")
+                field_value = f'{field}%3A%22{urllib.parse.quote(value)}%22'
+                field_values.append(field_value)
+            return ' AND '.join(field_values)
+        else:
+            field, value = search.split(':')
+            field = field.strip()
+            value = value.strip().strip("'")
+            field_value = f'{field}%3A%22{urllib.parse.quote(value)}%22'
+            return field_value
+    
+
+    
     @classmethod
     def add_endpoint(
         cls, search_operation: colrev.ops.search.Search, query: str
-    ) -> colrev.settings.SearchSource:
-        """Add SearchSource as an endpoint (based on query provided to colrev search -a )"""
-        raise NotImplementedError
+    ) -> typing.Optional[colrev.settings.SearchSource]:
+        """Add SearchSource as an endpoint (based on query provided to colrev search -a)"""
 
-    def run_search(
-        self, search_operation: colrev.ops.search.Search, rerun: bool
-    ) -> None:
-        """Run a search of ERIC"""
+        if "https://api.ies.ed.gov/eric/?" in query:
+            url_parsed = urllib.parse.urlparse(query)
+            query = urllib.parse.parse_qs(url_parsed.query)
+            search = query.get('search', [''])[0]
+            start = query.get('start', ['0'])[0]
+            rows = query.get('rows', ['2000'])[0]
+            if ':' in search:
+                search = ERICSearchSource.search_split(search)
+            filename = search_operation.get_unique_filename(
+                file_path_string=f"eric_{search}"
+            )
+            add_source = colrev.settings.SearchSource(
+                endpoint="colrev.eric",
+                filename=filename,
+                search_type=colrev.settings.SearchType.DB,
+                search_parameters={"query": search, 'start': start, 'rows': rows},
+                load_conversion_package_endpoint={"endpoint": "colrev.bibtex"},
+                comment="",
+            )
+            return add_source
+
+        return None
+
+    
+
+
+    def run_search(self, search_operation, rerun):
```

Comment:
Typing information is missing (see [pre-commit hooks](https://results.pre-commit.ci/run/github/363073613/1689000565.9wLAG3kWQPaH2eLhm5G0_Q)).